src/ebmc/ranking_function.cpp

```diff
@@ -114,51 +85,79 @@ ebmc_baset::propertyt &ranking_function_checkt::find_property()
   UNREACHABLE;
 }
 
-int ranking_function_checkt::operator()()
+int do_ranking_function(
+  const cmdlinet &cmdline,
+  message_handlert &message_handler)
 {
   // get the transition system
-  int exit_code = get_transition_system(
-    cmdline, message.get_message_handler(), transition_system);
+  transition_systemt transition_system;
+
+  int exit_code =
+    get_transition_system(cmdline, message_handler, transition_system);
 
   if(exit_code != -1)
     throw ebmc_errort();
 
   CHECK_RETURN(transition_system.trans_expr.has_value());
 
   // parse the ranking function
-  const auto ranking_function = parse_ranking_function();
+  if(!cmdline.isset("ranking-function"))
+    throw ebmc_errort() << "no candidate ranking function given";
+
+  const auto ranking_function = parse_ranking_function(
+    cmdline.get_value("ranking-function"), transition_system, message_handler);
 
   // find the property
-  exit_code = get_properties();
+  ebmc_propertiest properties;
 
-  if(exit_code != -1)
-    return exit_code;
+  if(ebmc_propertiest::from_transition_system(
+       transition_system, properties, message_handler))
+    throw ebmc_errort();
```

Comment:
Is an exception (without explanation) the right thing here? Perhaps use `CHECK_RETURN` instead?

Reply:
I'll change `ebmc_propertiest::from_transition_system` to throw its own exceptions.